### PR TITLE
IllegalArgumentException caused by historic data for ways

### LINKED ISSUE #422: IllegalArgumentException caused by historic data for ways
```
15:58:15.739 ERROR e.i.osmogrid.guardian.run.RunSupport - Node 268847628 of Way 24737158 is not within our nodes mapping
java.lang.IllegalArgumentException: Node 268847628 of Way 24737158 is not within our nodes mapping
	at utils.OsmoGridUtils$.$anonfun$buildStreetGraph$2(OsmoGridUtils.scala:126)
	at utils.OsmoGridUtils$.$anonfun$buildStreetGraph$2$adapted(OsmoGridUtils.scala:113)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:619)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:617)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1303)
	at utils.OsmoGridUtils$.$anonfun$buildStreetGraph$1(OsmoGridUtils.scala:113)
	at utils.OsmoGridUtils$.$anonfun$buildStreetGraph$1$adapted(OsmoGridUtils.scala:111)
	at scala.collection.immutable.VectorStatics$.foreachRec(Vector.scala:2124)
	at scala.collection.immutable.Vector.foreach(Vector.scala:2130)
	at utils.OsmoGridUtils$.buildStreetGraph(OsmoGridUtils.scala:111)
	at edu.ie3.osmogrid.lv.LvCoordinator$.$anonfun$awaitResults$1(LvCoordinator.scala:244)
	at org.apache.pekko.actor.typed.internal.BehaviorImpl$ReceiveBehavior.receive(BehaviorImpl.scala:146)
	at org.apache.pekko.actor.typed.Behavior$.interpret(Behavior.scala:283)
	at org.apache.pekko.actor.typed.Behavior$.interpretMessage(Behavior.scala:239)
	at org.apache.pekko.actor.typed.internal.adapter.ActorAdapter.handleMessage(ActorAdapter.scala:141)
	at org.apache.pekko.actor.typed.internal.adapter.ActorAdapter.aroundReceive(ActorAdapter.scala:117)
	at org.apache.pekko.actor.ActorCell.receiveMessage(ActorCell.scala:590)
	at org.apache.pekko.actor.ActorCell.invoke(ActorCell.scala:557)
	at org.apache.pekko.dispatch.Mailbox.processMailbox(Mailbox.scala:280)
	at org.apache.pekko.dispatch.Mailbox.run(Mailbox.scala:241)
	at org.apache.pekko.dispatch.Mailbox.exec(Mailbox.scala:253)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```

Only hypothesis, needs to be checked: This may be caused by bounderies that are only partly in the pbf-File and thus couldn't be of type ClosedWay.

Edit: Reason is historic data for ways.

### PR DESCRIPTION
**Pull request by danielfeismann**
resolves #422

### CODE REVIEW

File: src/main/scala/edu/ie3/osmogrid/model/OsmoGridModel.scala [hunk 1]
```diff
@@ -93,6 +106,42 @@
...
+              entity.metaInformation.map(_.version.getOrElse(0))
+            case _ => Some(0)
+          }
+          .reduceOption(_ max _)
+          .getOrElse(0)
+        entities.filter {
+          case entity: E if entity.metaInformation.isDefined =>
+            entity.metaInformation.exists(_.version.contains(maxVersion))
+          case _ => true
+        }
```
Comment (staudtMarius) [inline src/main/scala/edu/ie3/osmogrid/model/OsmoGridModel.scala:130]: Just one question from my side. Does this work if we have two versions of the same entity one with metaInformation and one without?
I think in this case, both versions are returned, which could cause problems. Maybe you could add a test for this case.

**Review by staudtMarius — CHANGES_REQUESTED** [review]
Just one question from my side.

File: src/main/scala/edu/ie3/osmogrid/model/OsmoGridModel.scala (hunk 1 above)
Comment (danielfeismann) [reply src/main/scala/edu/ie3/osmogrid/model/OsmoGridModel.scala:130]: Good point! I changed this into a logger warning and filter out these entities.

Comment (staudtMarius) [reply src/main/scala/edu/ie3/osmogrid/model/OsmoGridModel.scala]: What happens if we have an entity that occures only once but have no metaInformation? Will it be filtered out and cause a porblem later?

Comment (danielfeismann) [reply src/main/scala/edu/ie3/osmogrid/model/OsmoGridModel.scala]: As far as I checked there is always a version at the entity. So imho this is very unlikely. But sure we can split the entities and filter only those with multiple occurences. :)

Comment (staudtMarius) [reply src/main/scala/edu/ie3/osmogrid/model/OsmoGridModel.scala]: @danielfeismann Do you have an update on this part?

Comment (sebastian-peter) [reply src/main/scala/edu/ie3/osmogrid/model/OsmoGridModel.scala]: Honestly, OSM data that includes multiple versions of the same entity is just broken to me. I'd be fine with just assuming that the OSM data we're handling is coherent. Do we really need the meta information then?

Comment (danielfeismann) [reply src/main/scala/edu/ie3/osmogrid/model/OsmoGridModel.scala]: I agree with you, @sebastian-peter. Tried it again by using overpass, worked well at that location. So I'm fine with closing this one.

**Comment by danielfeismann** [comment]
Closed, since no longer necessary when using overpass.

**Pull request closed without merging**